core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -464,6 +465,11 @@ default ByteBufAllocator alloc() {
     @MustBeClosed
     SafeCloseable push();
 
+    @Override
+    default RequestContext unwrap() {
```

Comment:
`master` branch has been merged. Let's do the same work for `unwrapAll()`. 😉

Reply:
> Let's do the same work for unwrapAll()

Do you mean override the return type for `unwrapAll`?
I thought it's difficult to do this with the current constraints provided by `Unwrappable` 😅 

ref: https://github.com/line/armeria/pull/4338#discussion_r923111255

Reply:
Couldn't we safely cast the return type to `RequestContext`?
```
@Override
default RequestContext unwrapAll() {
    return (RequestContext) Unwrappable.super.unwrapAll();
}
```

Reply:
Sorry, I guess I should've made my comment on the other PR clearer 😅 

To be precise, as you said we can override the return type as `RequestContext` for objects that are "unwrappable".

However, I'm unsure how the API should be designed for objects that wrap an `Unwrappable` (i.e. via `AbstractUnwrappable`)

1. If `AbstractUnwrappable#unwrapAll` returns an `Object`:
In this case `AbstractUnwrappable#unwrapAll` returns an Object whereas `RequestContext#unwrapAll` returns a `RequestContext`, which causes a return type collision.
e.g.
```
'unwrapAll()' in 'com.linecorp.armeria.common.util.AbstractUnwrappable' clashes with 'unwrapAll()' in 'com.linecorp.armeria.common.RequestContext'; attempting to use incompatible return type
```

2. If `AbstractUnwrappable#unwrapAll` returns T:
I'm not sure if there's type safe way to know the innermost type of a wrappable.
```
final Foo foo = new Foo();
final Bar<Foo> bar = new Bar<>(foo);
final Qux<Bar<Foo>> qux = new Qux<>(bar);
```

ref: https://github.com/line/armeria/pull/4338#discussion_r914680171

Let me know if you have a better idea though 😅

Reply:
> 'unwrapAll()' in 'com.linecorp.armeria.common.util.AbstractUnwrappable' clashes with 'unwrapAll()' in 'com.linecorp.armeria.common.RequestContext'; attempting to use incompatible return type

The error seems to happen in `RequestContextWrapper`. How about overriding the return type of `unwrapAll()` in `RequestContextWrapper`?
I don't see that we need to change the return type of `AbstractUnwrappable#unwrapAll`.
The subtype of `AbstractUnwrappable` or `Unwrappable` freely overrides the return type if necessary.
```java
public abstract class RequestContextWrapper<T extends RequestContext>
        extends AbstractUnwrappable<T> implements RequestContext {

    @Override
    public RequestContext unwrapAll() {
        return (RequestContext) super.unwrapAll();
    }
}
```

Reply:
Ah, I think you're right. I guess we can override the return type for wrapper types only. Done!